packages/ckeditor5-editor-multi-root/src/multirooteditor.ts

```diff
@@ -123,6 +126,22 @@ export default class MultiRootEditor extends DataApiMixin( Editor ) {
 		const view = new MultiRootEditorUIView( this.locale, this.editing.view, rootNames, options );
 
 		this.ui = new MultiRootEditorUI( this, view );
+
+		// This can be improved if `$isVirtual` attribute for root is introduced.
+		// Multi-root editor will create (or not) the editable based on the attribute value. The events will be editable-related.
+		this.model.document.on( 'change:data', () => {
+			const changedRoots = this.model.document.differ.getChangedRoots();
+
+			for ( const [ rootName, isAttached ] of changedRoots ) {
+				const root = this.model.document.getRoot( rootName )!;
+
+				if ( isAttached ) {
+					this.fire<AddRootEvent>( 'addRoot', root );
+				} else {
+					this.fire<DetachRootEvent>( 'detachRoot', root );
+				}
+			}
+		} );
```

Comment:
Next step of multi-root project will include work on roots attributes. I would like to introduce an attribute that would be handled only by the multi-root editor. If the attribute is set, multi-root editor will automatically create or remove editable instance if such root was not virtual (the usual case). After the change is introduced, multi-root editor will fire `addEditable` and `removeEditable` events instead and the integrator will only need to append/remove the DOM element. It will be easier to understand.